packages/cletus/src/agents/chat-orchestrator.ts

```diff
@@ -284,22 +307,40 @@ export async function runChatOrchestrator(
               break;
 
             case 'responseTokens':
-              setOutputTokens(chunk.tokens);
+              // Add as estimated text output tokens
               pending.tokens = chunk.tokens;
+              if (!currentUsage.text) {
+                currentUsage.text = {};
+              }
+              currentUsage.text.output = chunk.tokens;
+              updateUsageEvent();
               break;
 
             case 'reason':
-              setReasoningTokens(Math.ceil(chunk.content.length / 4));
+              // Add reasoning token estimation to current usage
+              const reasoningTokens = Math.ceil(chunk.content.length / 4);
+              if (!currentUsage.reasoning) {
+                currentUsage.reasoning = {};
+              }
+              currentUsage.reasoning.output = reasoningTokens;
+              updateUsageEvent();
               break;
 
             case 'toolStart':
-              addDiscardedTokens(Math.ceil((chunk.tool.name.length + JSON.stringify(chunk.args).length) / 4));
+              // Add to estimated output tokens for tool calls
+              const toolTokens = Math.ceil((chunk.tool.name.length + JSON.stringify(chunk.args).length) / 4);
+              if (!currentUsage.text) {
+                currentUsage.text = {};
+              }
+              currentUsage.text.output = (currentUsage.text.output || 0) + toolTokens;
```

Comment:
This would get cleared out by the next                 updateEstimatedTextOutputTokens() call. We should have a toolTokens number that this increments and that function adds toolTokens to the output tokens.